src/domains/login/login.container.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import { APP_PATH } from 'routes';
+import { LoginComponent } from './login.component';
+
+export const Login = () => {
+  const isValidSession = useSelector((state) =>
+    state.app.get('isValidSession'),
+  );
+
+  return isValidSession ? <Redirect to={APP_PATH} /> : <LoginComponent />;
```

Comment:
Question, can this logic check go into the router? Like check the state there and make the redirection there, 
i.e. 
`<Route exact path="/login">
  {isValidSession ? <Redirect to={`/${APP_PATH}`} /> : <Login />}
</Route>`

Reply:
Even better would be something like `isLoggedIn` and the dependencies for that would be a valid token + a valid session

Reply:
Is there a benefit to doing that? 🤔 

IMO it seems hard to read xD

Reply:
The benefit is that when you look for instance for an issue with redirection, it makes more sense to look for it in the router than it makes to go look for it in a specific component, besides its the router responsibility to tell the app where to go and under what conditions

Reply:
True, makes more sense